include/oneapi/dpl/pstl/hetero/dpcpp/sycl_traits.h

```diff
@@ -0,0 +1,104 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+// This file contains some specialization SYCL traits for some oneDPL types
+//
+// Include this header before a kernel submit SYCL code
+
+#ifndef _ONEDPL_SYCL_TRAITS_H
+#define _ONEDPL_SYCL_TRAITS_H
+
+#if __INTEL_LLVM_COMPILER && (__INTEL_LLVM_COMPILER < 20240100)
+
+#define _ONEDPL_DEVICE_COPYABLE(TYPE) \
+template<typename... Ts > \
+struct sycl::is_device_copyable<TYPE<Ts...>, ::std::enable_if_t<!std::is_trivially_copyable_v<TYPE<Ts...>>>>: ::std::conjunction<sycl::is_device_copyable<Ts>...> {};
+
+#else
+
+#define _ONEDPL_DEVICE_COPYABLE(TYPE) \
+template<typename... Ts > \
+struct sycl::is_device_copyable<TYPE<Ts...>>: ::std::conjunction<sycl::is_device_copyable<Ts>...> {};
+
+#endif
+
+using namespace oneapi::dpl::__internal;
+
+_ONEDPL_DEVICE_COPYABLE(__not_pred)
```

Comment:
Do we need to include the headers with all the definitions of these functors / iterators?

Reply:
Actually, it supposed to be included in file where a submit is called, f.e.  `paralle_backend_sycl.h.`
it means the all mentioned types are known to a compiler.
BTW, I've really missed including that file.. I will fix it.

Reply:
Do we need it to be included also in other submit locations like `parallel_backend_sycl_*.h`, (radix_sort, radix_sort_one_wg, reduce, histogram) ?

Reply:
Probably, it makes sense to be included into "sycl_defs.h" once ?

Reply:
> Actually, it supposed to be included in file where a submit is called, f.e. paralle_backend_sycl.h.

Sorry, but that's a hack... It is not reliable and does not work for kernel templates, which do not include `parallel_backend_sycl.h`.